src/st/lib_psp/unk_1D9E8.c

```diff
@@ -5,7 +5,109 @@ INCLUDE_ASM("st/lib_psp/psp/lib_psp/unk_1D9E8", func_psp_09255068);
 
 INCLUDE_ASM("st/lib_psp/psp/lib_psp/unk_1D9E8", func_us_801BB8DC);
 
-INCLUDE_ASM("st/lib_psp/psp/lib_psp/unk_1D9E8", func_psp_09256018);
+// func_us_801BBAB4 code is shared with PSX and is ready for merge
+// Seems to be the windup just before the spit attack
+u8 func_us_801BBAB4(void) {
+    Primitive* prim;
+    Pos tempPrim;
+    s32 primIndex;
+    s32 unkVar;
+    u8 randomVal;
+    u8 ret;
+#ifdef VERSION_US
+    s32 dummy[2];
+#endif
+
+    ret = false;
+    switch (g_CurrentEntity->ext.lesserDemon.unk84) {
+    case 0:
+        primIndex = g_api.AllocPrimitives(PRIM_TILE, 0x18);
+        if (primIndex != -1) {
+            g_CurrentEntity->primIndex = primIndex;
+            g_CurrentEntity->flags |= FLAG_HAS_PRIMS;
+            prim = &g_PrimBuf[primIndex];
+            g_CurrentEntity->ext.prim = prim;
```

Comment:
Presumably `g_CurrentEntity` is the lesser demon here? So I think all these `g_CurrentEntity->ext.prim` probably make sense updated to `g_CurrentEntity->ext.lesserDemon.unk7C` instead

Reply:
Yes, and I agree now that I know how they're connected.